Lib/test/test_print.py

```diff
@@ -155,7 +155,7 @@ def test_string_with_excessive_whitespace(self):
 
         self.assertIn('print("Hello World", end=" ")', str(context.exception))
 
-    def test_string_with_stream_redirection(self):
+    def test_print_string_with_stream_redirection(self):
         import sys
         python2_print_str = 'print >> sys.stderr, "message"'
```

Comment:
Yes, that's what I'm saying - the code that checked for the custom `SyntaxError` needed to be tested with `exec`, because putting it outside a string would cause the entire module to fail to compile. That isn't the case here, so you can just do `print >> sys.stderr, "message"` directly in the body of the context manager (and similarly for the other cases).